tests/testthat/test-sparsevctrs.R

```diff
@@ -120,3 +120,51 @@ test_that("sparse matrices can be passed to `fit() - xy", {
   # We expect 1 materialization - the outcome
   expect_snapshot(wf_fit <- fit(wf_spec, hotel_data))
 })
+
+test_that("sparse tibble can be passed to `predict()", {
+  skip_if_not_installed("glmnet")
+  # Make materialization of sparse vectors throw an error
+  # https://r-lib.github.io/sparsevctrs/dev/reference/sparsevctrs_options.html
+  withr::local_options("sparsevctrs.verbose_materialize" = 3)
+
+  hotel_data <- sparse_hotel_rates(tibble = TRUE)
+
+  spec <- parsnip::linear_reg(penalty = 0) %>%
+    parsnip::set_mode("regression") %>%
+    parsnip::set_engine("glmnet")
+
+  rec <- recipes::recipe(avg_price_per_room ~ ., data = hotel_data)
+
+  wf_spec <- workflow() %>%
+    add_recipe(rec) %>%
+    add_model(spec)
+
+  wf_fit <- fit(wf_spec, hotel_data)
+  
+  expect_no_error(predict(wf_fit, hotel_data))
+})
+
+test_that("sparse matrix can be passed to `predict()", {
+  skip_if_not_installed("glmnet")
+  # Make materialization of sparse vectors throw a message
+  # https://r-lib.github.io/sparsevctrs/dev/reference/sparsevctrs_options.html
+  withr::local_options("sparsevctrs.verbose_materialize" = 1)
+
+  hotel_data <- sparse_hotel_rates()
+
+  spec <- parsnip::linear_reg(penalty = 0) %>%
+    parsnip::set_mode("regression") %>%
+    parsnip::set_engine("glmnet")
+
+  rec <- recipes::recipe(avg_price_per_room ~ ., data = hotel_data)
+
+  wf_spec <- workflow() %>%
+    add_recipe(rec) %>%
+    add_model(spec)
+
+  expect_snapshot(
+    wf_fit <- fit(wf_spec, hotel_data)
+  )
```

Comment:
```suggestion
  wf_fit <- fit(wf_spec, hotel_data)
```

This snapshot is a duplicate from above